src/usr/local/buildpack/utils/filesystem.sh

```diff
@@ -44,13 +44,17 @@ function setup_directories () {
 
   mkdir -p "${install_dir}"
   # contains the installed tools
-  mkdir -m 770 "$(get_tools_path)"
+  # shellcheck disable=SC2174
+  mkdir -p -m 770 "$(get_tools_path)"
   # contains env for the installed tools
-  mkdir -m 770 "$(get_env_path)"
+  # shellcheck disable=SC2174
+  mkdir -p -m 770 "$(get_env_path)"
   # contains the latest version of the tools
-  mkdir -m 770 "$(get_version_path)"
+  # shellcheck disable=SC2174
+  mkdir -p -m 770 "$(get_version_path)"
   # contains the wrapper and symlinks for the tools
-  mkdir -m 770 "$(get_bin_path)"
+  # shellcheck disable=SC2174
+  mkdir -p -m 755 "$(get_bin_path)"
```

Comment:
Does it change permission when directory exist? As `/usr/local/bin` already exist.

Reply:
It does not, the test failed before when I tested for `770`. So the `-p` checks if the directory exist and if it does simply returns.

Reply:
Ok, so we need to make sure it's writable by root group, so we need `775` on it.